src/Listeners/Cron.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Endurance\WP\Module\Data\Listeners;
+
+/**
+ * Schedules Cron event listeners
+ */
+class Cron extends Listener {
+
+	/**
+	 * Register all required hooks for the listener category
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+
+		// Ensure there is a weekly option in the cron schedules
+		add_filter( 'cron_schedules', array( $this, 'add_weekly_schedule' ) );
+
+		// Weekly cron hook
+		add_action( 'bh_data_cron', array( $this, 'update' ) );
+
+		// Register the cron task
+		if ( ! wp_next_scheduled( 'bh_data_cron' ) ) {
```

Comment:
If the cron is not scheduled, would we want it to run sooner than next week? Maybe `time() + MINUTE_IN_SECONDS`?

Reply:
It doesn't really matter when it runs the first time. We're mostly just adding in case there is no admin traffic on the site for a while going forward. 

If a site is connecting for the first time, then we don't need to run it right away since the version data is sent with the connection, but if it's already been connected in the past and updates the plugin to this version, I guess it wouldn't hurt to fire it a little sooner. 

I'll drop it to `time() + DAY_IN_SECONDS` since we don't need to fire it right away for those brand new connections.